### ISSUE TITLE
Unable to get rootless Podman to work with NFS storage

### ISSUE THREAD

**Issue by Deliveranc3** [containers/podman#22002]
### Discussed in https://github.com/containers/podman/discussions/21606

<div type='discussions-op-text'>

<sup>Originally posted by **Deliveranc3** February 11, 2024</sup>
I've been pulling my hair out for days on this. I have a Synology NAS with an NFS share that I want to use as storage for rootless Podman containers on a Fedora VM. 

I've mounted the NFS share on the Fedora VM and created a directory owned by my rootless user. This is all fine, except when I run a container that tries to chmod files to a uid within its user namespace, the NFS server denies the operation which causes the container to fail. I'm using a bind volume mount for the container.

I came across [this](https://www.redhat.com/sysadmin/podman-rootless-overlay) article which includes the following:

> One prominent feature we have added is support for storing file security attributes in [extended file attributes](https://en.wikipedia.org/wiki/Extended_file_attributes) (xattrs). We need this to support NFS home directories. NFS servers block the use of containers with more than one UID within the user namespace. ... When a process running in this mode creates a file with a different UID, fuse-overlay intercepts the UID creation, creates the file with the mounters UID, and stores the different UID in an xattr.

This sounds like exactly what I need, so I created `~/.config/containers/storage.conf` and changed mount_program to `/usr/bin/fuse-overlayfs`. I did a `podman system reset` then verified that `overlay.mount_program` is set to fuse-overlayfs in `podman info`. I don't see any difference in behavior, however. I still get permission denied with bind mounts. I tried to create and mount a podman volume as described [here](https://access.redhat.com/solutions/6415411), but I still get permission denied. I don't see any indication that podman is actually using fuse. Am I doing something wrong?

I'm on Fedora 38, kernel 6.7.4-100.fc38.x86_64, and podman 4.8.3
</div>

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
does your NFS mount support extended file attributes?

From the discussion, I see you get the error `fuse-overlayfs: read xattr `user.fuseoverlayfs.override_stat` from upperdir: Operation not supported`.


That means fuse-overlayfs is not able to store the correct file mode/ownership, so you can't really use this mount inside a user namespace.

**Comment by Deliveranc3** [comment]
I don't believe my NFS server supports xattrs. Maybe I'm misunderstanding fuse-overlayfs. I thought that (one of) its purposes was to enable the use of xattrs on backing storage which didn't support it, by way of storing those xattrs in a separate file?

**Comment by jiridanek** [comment]
> I thought that (one of) its purposes was to enable the use of xattrs on backing storage which didn't support it, by way of storing those xattrs in a separate file?

@Deliveranc3 you're confusing two different fuse filesystems. The [fuse-overlayfs](https://github.com/containers/fuse-overlayfs) is what giuseppe says it is, and what you are describing a _different_ filesystem called [fuse_xattrs](https://github.com/fbarriga/fuse_xattrs).
